Add bao operator subcommand for config validation (#1609)
openbao/openbao: changelog/1609.txt

> ```release-note:feature

Reviewer: This could probably just be an improvement :-)

Author: done

> cli: add new subcommand "bao operator validate-config" to validate a config.
> It is similar to "bao operator diagnose", but it only validates the config and not the system it is running on (e.g. it will not check that the raft directory is writable).
> This allows it to be used on an operators Laptop or in CI validations.

Reviewer: ```suggestion
cli: add new subcommand "bao operator validate-config" to validate a configuration file syntax
```

Only features are multi-line.
